src/client/datascience/interactive-ipynb/nativeEditor.ts

```diff
@@ -713,26 +718,46 @@ export class NativeEditor extends InteractiveBase implements INotebookEditor {
             // Make sure to clear so we don't use this again.
             this.localStorage.update(key, undefined);
 
-            // Transfer this to global storage so we use that next time instead
-
-            const stat = await this.fileSystem.stat(this.file.fsPath);
-            this.globalStorage.update(key, { contents: workspaceData, lastModifiedTimeMs: stat ? stat.mtime : undefined });
+            // Transfer this to a file so we use that next time instead.
+            const filePath = this.getHashedFileName(key);
+            await this.writeToStorage(filePath, workspaceData);
 
             return workspaceData;
         }
     }
 
-    // This is what we could have done to transferFromGlobalStorage all of the old keys
     // VS code recommended we use the hidden '_values' to iterate over all of the entries in
-    // the global storage map.
-    // private async transferFromGlobalStorage(): Promise<void> {
-    //     if ((this.globalStorage as any)._values) {
-    //         const map = (this.globalStorage as any)._values as Map<string, any>;
-    //         [...map.keys()].forEach(k => {
-    //             if (k.startsWith('notebook-'))
-    //         })
-    //     }
-    // }
+    // the global storage map and delete the ones we own.
+    private async transferStorage(): Promise<void[]> {
+        const promises: Thenable<void>[] = [];
+
+        try {
+            // tslint:disable-next-line: no-any
+            if ((this.globalStorage as any)._value) {
+                // tslint:disable-next-line: no-any
+                const keys = Object.keys((this.globalStorage as any)._value);
+                [...keys].forEach((k: string) => {
+                    if (k.startsWith(KeyPrefix)) {
+                        // Write each pair to our alternate storage, but don't bother waiting for each
+                        // to finish.
+                        const filePath = this.getHashedFileName(k);
+                        const contents = this.globalStorage.get(k);
+                        if (contents) {
+                            this.writeToStorage(filePath, JSON.stringify(contents)).ignoreErrors();
+                        }
+
+                        // Remove from the map so that global storage does not have this anymore.
+                        // Use the real API here as we don't know how the map really gets updated.
+                        promises.push(this.globalStorage.update(k, undefined));
+                    }
+                });
+            }
+        } catch (e) {
+            traceError(`Exception eliminating global storage parts: ${e}`);
```

Comment:
This way, the exception object isn't converted to to string and we don't lose information.
Else we could end up loosing stack trace, etc and just `[Object object]` being printed out.
```suggestion
            traceError(`Exception eliminating global storage parts`, e);
```